internal/controllers/topology/cluster/patches/template.go

```diff
@@ -57,11 +57,14 @@ func (t *requestItemBuilder) WithHolder(object client.Object, fieldPath string)
 	return t
 }
 
+// uuidGenerator is defined as a package variable to enable changing it during testing.
+var uuidGenerator func() types.UID = uuid.NewUUID
```

Comment:
This is one of the things I'm not fully comfortable with - it lets us generate the UUID at runtime for testing. It's fully private, but it's not being injected properly. The alternative would be to pass it through a call chain though as this is essentially for matching patches.

Reply:
I'm fine with the current implementation and prefer it over alternatives. I've seen this pattern used before and I don't think it's too problematic.